Airbrakenotifier notify method uses synchronic Http calls

sinanselimoglu opened the pull request:
#44 I've changed the name of the notify method as notifySync and added a new method named notifyAsync. Now exception mappers/handlers do not need to wait for airbrake http connection.

Sorry for the indentation.

Changes line =  47-58 NotifySync method / 60-68 NotifyAsync method

Commit:
Airbrakenotifier notify method uses synchronic Http calls

sinanselimoglu commented:
#44  I've changed the name of the notify method as notifySync and added a new method named notifyAsync. Now exception mappers/handlers do not need to wait for airbrake http connection.

Sorry for the indentation.

jvosloo commented:
I would suggest you leave the current method “notify” for backwards compatibility.
Introduce the 2 new methods, with the “notifySync” method merely being wrapped by “notify”.


Or better yet, leave “notify as default, synchronous method as it is currently, and add the notifySync method.


Just my 2 cents




> On Mar 13, 2016, at 8:31 PM, Sinan <notifications@github.com> wrote:
> 
> 
> #44
> I've changed the name of the notify method as notifySync and added a new method named notifyAsync. Now exception mappers/handlers do not need to wait for airbrake http connection.
> 
> Sorry for the indentation.
> 
> —
> Reply to this email directly or view it on GitHub
> .<embedded image>
> 
>

Commit:
Airbrakenotifier notify method uses synchronic Http calls

sinanselimoglu commented:
Hey @jvosloo 

Thanks for your feedback, I updated pr like you said. Just added notifyAysnc method uses normal notify.

However Intelij forced me to add this line;

`final AirbrakeNotifier airbrakeNotifier = this;`

Because notify method mess with object's notify method and it doesn't compile.(IDE says you can not pass parameters to notify method.) I used "this.notify"  but it doesn't help. Only when I create another final variable and assign it to _**this**_ , then it worked.

How is now ?

vmihailenco commented:
There is new notifier at https://github.com/airbrake/javabrake that supports async API. This notifier is de-facto not maintained properly and deprecated.
